### PR TITLE
include cdf method in distribution

### PR DESCRIPTION
**Pull request by vafl**
*Issue #, if available:*

*Description of changes:*

This includes a `cdf` method for (some of) the distribution.
Also fixes a couple of bugs in distributions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-120/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/1/index.html

**Comment by szha** [comment]
Job PR-120/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/2/index.html

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=h1) Report
> Merging [#120](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/e86091520ae8d2319dade1f4fa7fc6d95cacc97d?src=pr&el=desc) will **decrease** coverage by `0.21%`.
> The diff coverage is `75.71%`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #120      +/-   ##
==========================================
- Coverage   79.05%   78.84%   -0.22%     
==========================================
  Files         110      110              
  Lines        6301     6366      +65     
==========================================
+ Hits         4981     5019      +38     
- Misses       1320     1347      +27
```

| [Impacted Files](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/gluonts/distribution/laplace.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2xhcGxhY2UucHk=) | `95.65% <100%> (-4.35%)` | :arrow_down: |
| [...c/gluonts/distribution/transformed\_distribution.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3RyYW5zZm9ybWVkX2Rpc3RyaWJ1dGlvbi5weQ==) | `93.84% <100%> (+0.86%)` | :arrow_up: |
| [src/gluonts/distribution/binned.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2Jpbm5lZC5weQ==) | `91.66% <100%> (-3.21%)` | :arrow_down: |
| [src/gluonts/distribution/gaussian.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2dhdXNzaWFuLnB5) | `90.19% <100%> (-1.11%)` | :arrow_down: |
| [src/gluonts/support/util.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvc3VwcG9ydC91dGlsLnB5) | `68.51% <30.76%> (-5.17%)` | :arrow_down: |
| [src/gluonts/distribution/distribution.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2Rpc3RyaWJ1dGlvbi5weQ==) | `79.1% <50%> (-0.9%)` | :arrow_down: |
| [src/gluonts/distribution/uniform.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3VuaWZvcm0ucHk=) | `86.95% <66.66%> (-3.96%)` | :arrow_down: |
| [src/gluonts/distribution/mixture.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL21peHR1cmUucHk=) | `96.47% <72.72%> (-0.9%)` | :arrow_down: |
| [src/gluonts/distribution/bijection.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2JpamVjdGlvbi5weQ==) | `77.67% <82.35%> (+0.83%)` | :arrow_up: |
| [src/gluonts/distribution/student\_t.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3N0dWRlbnRfdC5weQ==) | `94.44% <0%> (-5.56%)` | :arrow_down: |
| ... and [4 more](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree-more) | |

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=h1) Report
> Merging [#120](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/gluon-ts/commit/e86091520ae8d2319dade1f4fa7fc6d95cacc97d?src=pr&el=desc) will **decrease** coverage by `0.21%`.
> The diff coverage is `75.71%`.


[Impacted file tree graph](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #120      +/-   ##
==========================================
- Coverage   79.05%   78.84%   -0.22%     
==========================================
  Files         110      110              
  Lines        6301     6366      +65     
==========================================
+ Hits         4981     5019      +38     
- Misses       1320     1347      +27
```

| [Impacted Files](https://codecov.io/gh/awslabs/gluon-ts/pull/120?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/gluonts/distribution/laplace.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2xhcGxhY2UucHk=) | `95.65% <100%> (-4.35%)` | :arrow_down: |
| [...c/gluonts/distribution/transformed\_distribution.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3RyYW5zZm9ybWVkX2Rpc3RyaWJ1dGlvbi5weQ==) | `93.84% <100%> (+0.86%)` | :arrow_up: |
| [src/gluonts/distribution/binned.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2Jpbm5lZC5weQ==) | `91.66% <100%> (-3.21%)` | :arrow_down: |
| [src/gluonts/distribution/gaussian.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2dhdXNzaWFuLnB5) | `90.19% <100%> (-1.11%)` | :arrow_down: |
| [src/gluonts/support/util.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvc3VwcG9ydC91dGlsLnB5) | `68.51% <30.76%> (-5.17%)` | :arrow_down: |
| [src/gluonts/distribution/distribution.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2Rpc3RyaWJ1dGlvbi5weQ==) | `79.1% <50%> (-0.9%)` | :arrow_down: |
| [src/gluonts/distribution/uniform.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3VuaWZvcm0ucHk=) | `86.95% <66.66%> (-3.96%)` | :arrow_down: |
| [src/gluonts/distribution/mixture.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL21peHR1cmUucHk=) | `96.47% <72.72%> (-0.9%)` | :arrow_down: |
| [src/gluonts/distribution/bijection.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL2JpamVjdGlvbi5weQ==) | `77.67% <82.35%> (+0.83%)` | :arrow_up: |
| [src/gluonts/distribution/student\_t.py](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree#diff-c3JjL2dsdW9udHMvZGlzdHJpYnV0aW9uL3N0dWRlbnRfdC5weQ==) | `94.44% <0%> (-5.56%)` | :arrow_down: |
| ... and [4 more](https://codecov.io/gh/awslabs/gluon-ts/pull/120/diff?src=pr&el=tree-more) | |

File: test/distribution/test_mixture.py [hunk 1]
```diff
@@ -39,7 +40,7 @@ def diff(x: NPArrayLike, y: NPArrayLike) -> np.ndarray:
     return np.mean(np.abs(x - y))
 
 
-NUM_SAMPLES = 1000
+NUM_SAMPLES = 100_000
```
Comment (geoalgo) [inline test/distribution/test_mixture.py]: does the test still run in less than 2s?

**Comment by szha** [comment]
Job PR-120/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/3/index.html

File: src/gluonts/distribution/bijection.py [hunk 2]
```diff
@@ -68,6 +68,10 @@ def inverse_bijection(self) -> 'Bijection':
     def event_dim(self) -> int:
         raise NotImplementedError()
 
+    @property
+    def sign(self) -> Tensor:
```
Comment (geoalgo) [inline src/gluonts/distribution/bijection.py]: Can you add a comment describing the contract of the method?

**Comment by szha** [comment]
Job PR-120/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/4/index.html

File: test/distribution/test_mixture.py (hunk 1 above)
Comment (vafl) [reply test/distribution/test_mixture.py]: I made some changes and now it runs in under 2s.

File: src/gluonts/distribution/bijection.py (hunk 2 above)
Comment (vafl) [reply src/gluonts/distribution/bijection.py]: done

**Comment by szha** [comment]
Job PR-120/5 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/5/index.html

**Comment by szha** [comment]
Job PR-120/6 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/6/index.html

File: src/gluonts/distribution/uniform.py [hunk 3]
```diff
@@ -81,12 +81,15 @@ def s(low: Tensor, high: Tensor) -> Tensor:
             raw_samples = self.F.sample_uniform(
                 low=low.zeros_like(), high=high.ones_like()
             )
-            return low + raw_samples + high
+            return low + raw_samples * (high - low)
```
Comment (lostella) [inline src/gluonts/distribution/uniform.py]: Apparently we lack some testing here :-)

**Comment by szha** [comment]
Job PR-120/7 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/7/index.html

**Comment by szha** [comment]
Job PR-120/8 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/8/index.html

**Comment by szha** [comment]
Job PR-120/9 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-120/9/index.html

**Pull request merged**